src/api/app/models/user.rb

```diff
@@ -696,16 +700,7 @@ def delete!
   end
 
   def involved_projects_ids
-    # just for maintainer for now.
-    role = Role.hashed['maintainer']
-
-    ### all projects where user is maintainer
-    projects = relationships.projects.where(role_id: role.id).pluck(:project_id)
-
-    # all projects where user is maintainer via a group
-    projects += Relationship.projects.where(role_id: role.id).joins(:groups_users).where(groups_users: { user_id: id }).pluck(:project_id)
-
-    projects.uniq
+    (direct_maintained_projects.pluck(:project_id) + group_maintained_projects.pluck(:project_id)).uniq
```

Comment:
what about using `||` instead of `+`, then you don't need `uniq`:

`direct_maintained_projects.pluck(:project_id) || group_maintained_projects.pluck(:project_id)`